src/main/java/org/jabref/logic/exporter/TemplateExporter.java

```diff
@@ -172,7 +171,7 @@ private Reader getReader(String filename) throws IOException {
         String name = dir + filename;
         Reader reader;
         // Try loading as a resource first. This works for files inside the JAR:
-        URL reso = JabRefMain.class.getResource(name);
+        URL reso = TemplateExporter.class.getResource(name);
```

Comment:
Now `name` needs to be relative to `org/jabref/logic/exporter/` and not `org/jabref/` right?

Reply:
I checked the code and tried out the UI:

![grafik](https://user-images.githubusercontent.com/1366654/38172104-65eb9076-35a6-11e8-9ec2-d47c1b9d8339.png)

The UI works and the code IMHO never goes that path. With that change, the code is prepared for testing the "TemplateExporter".

Reply:
Sorry, but I don't really get what you want to say. You tested it and it worked although the changed code was never invoked?

https://github.com/JabRef/jabref/blob/7844fcbbea65bcbbe6eaf624352ebbe76d897946/src/main/java/org/jabref/logic/exporter/TemplateExporter.java#L161-L171
Gives a name of the form `/resource/layout/<exporter name>.layout`.  With the above change, I suspect that all layout files in `src/main/resources/resource/layout` are no longer found and thus the corresponding exporter stop to work.

Reply:
TL;DR: The given name is **absolute**, so it does not matter which class is used.

Longer reasoning (with debugging JabRef):

JabRefs main class is in `org.jabref`, so the path changes from

    file:/C:/git-repositories/jabref/jabref/out/production/resources/resource/layout/html.begin.layout

to

    file:/C:/git-repositories/jabref/jabref/out/production/resources/resource/layout/html.begin.layout

So, no change when debugging JabRef.

I think, it is because the parameter `directory` is given and thus the path is an **absolute** one. (Source: https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getResource(java.lang.String))

I would like to change the path to enable the test files residing in the test `resources` folder of "TemplateExporter".

Reply:
Ok, if it works then the change is of course fine with me. I was only concerned that the exporter may break.